### PR TITLE
Adds ruby 3.2 to test matrix.

### PR DESCRIPTION
**Pull request by LeFnord**
- removes unsupported ones from it

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
-on: [push, pull_request]
+
+on:
+  push:
+    branches:
+      - '*'
+  pull_request:
+    branches:
+      - '*'
+
```
Comment (kubo) [inline .github/workflows/tests.yml:10]: Could you explain the reason to change here?

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
 jobs:
   tests:
     continue-on-error: ${{ matrix.experimental }}
     strategy:
       fail-fast: false
       matrix:
-        ruby: [2.5, 2.6, 2.7, '3.0', 3.1]
+        ruby: [2.7, 3.0, 3.1, 3.2]
```
Comment (kubo) [inline .github/workflows/tests.yml:17]: Could you revert `3.0` to `'3.0'`?

It is intentional. When `matrix.ruby` is number `3.0`, `${{ matrix.ruby }}` is converted to `3` and the latest version of ruby 3.x is picked up. See https://github.com/LeFnord/ruby-oci8/actions/runs/4222066763/jobs/7330111735#step:7:3 and https://github.com/LeFnord/ruby-oci8/actions/runs/4222066763/jobs/7330111735#step:7:26.
It must be string `'3.0'` to use the latest version of ruby 3.0.x.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (LeFnord) [reply .github/workflows/tests.yml:10]: with that setup, I can run the action on my fork, to check if it is all fine before making a PR

see → https://github.com/LeFnord/ruby-oci8/actions

File: .github/workflows/tests.yml (hunk 2 above)
Comment (LeFnord) [reply .github/workflows/tests.yml:17]: ah … sure, will revert it … and yeah it is really a strange behaviour, thanks for spotting it

**Comment by kubo** [comment]
I merged this pull request using command line, not by using "Merge pull request" button in this page.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (kubo) [reply .github/workflows/tests.yml:10]: I excluded this change from the commit https://github.com/kubo/ruby-oci8/commit/b9d83fcb707899ceaa16e8a7d26c233735a77d73 merged on the master branch.

**Pull request closed without merging**